### DISCUSSION TITLE
How to decode protobuf in Node Red?

### DISCUSSION THREAD

**Discussion by martinwozenilek** [meshtastic/Discussions#51] [nRF52]
When I try to decode the protobuf messages from a nRF52 device in Node Red I get this error message: 

`Wire format is invalid: Error: no such Type or Enum 'MeshPacket' in Type .meshtastic.ServiceEnvelope`

For decoding I'm using **node-red-contrib-protobuf in version 1.1.2**. Some other users had the same problem but never get a working answer in the old discourse group. 

And by the way: will json format come to the nRF52 devices or is this completely impossible? This is very important for me as I use the Mesh mainly for telemetry data.

**Comment by martinwozenilek** [comment]
Closed. I've recognized that my Meshtastic MQTT gateway, which is Raspberry Pi based, does the conversion from protobuf to json already. This way I can use my nRF52 devices for telemetry data.

**Comment by JohnathonMohr** [comment]
For others who come across this and are struggling to get **node-red-contrib-protobuf** to work like I was, I was able to figure this out:

The Proto File must be a **comma-separated list of every .proto file needed to deserialize a payload** (not just the single file that contains the definition of the message type being parsed).

For example, if your Meshtastic protobuf files are in /data/meshtastic/, the **Path** would need to be set to:
`/data/meshtastic/admin.proto,/data/meshtastic/apponly.proto,/data/meshtastic/atak.proto,/data/meshtastic/cannedmessages.proto,...`

This line will be very verbose, but after listing out every single protobuf definition file, the decode node works. (I tried specifying */data/meshtastic/\*.proto*, but it didn't work with a wildcard.)

If running in Linux, here's a quick bash command to make generating that list easier:
`find /path/to/protobufs -name "*.proto" -print0 | tr '\0' ',' | sed 's/,$/\n/'`

Note that, if running Node Red in a container, the files will be mapped differently inside the container, so the paths in the string will need to be updated accordingly.
